dask/base.py

```diff
@@ -1236,19 +1239,28 @@ def _normalize_pure_object(o: object) -> tuple[str, int]:
 
 def _normalize_pickle(o: object) -> tuple:
     buffers: list[pickle.PickleBuffer] = []
-    pik: bytes | None
-    try:
-        pik = pickle.dumps(o, protocol=5, buffer_callback=buffers.append)
-        if b"__main__" in pik:
-            pik = None
-    except Exception:
-        pik = None
+    pik: int | None = None
+    pik2: int
 
+    for mod in [pickle, cloudpickle]:
+        for _ in range(3):
```

Comment:
this makes me a little nervous but is the only way I found to actually ensure that the output is the same. there is still some hidden mutation that I couldn't quite track down